test/e2e/traceflow_test.go

```diff
@@ -1468,7 +1465,7 @@ func testTraceflowInterNode(t *testing.T, data *TestData) {
 						{
 							Component:       v1alpha1.ComponentLB,
 							Pod:             fmt.Sprintf("%s/%s", testNamespace, nginxPodName),
-							TranslatedSrcIP: "169.254.169.252",
+							TranslatedSrcIP: gatewayIPv4,
```

Comment:
I'm not very sure which source IP makes more sense for hairpin case, a virtual IP or the gateway IP. Maybe it's not very important. But we should at least mention the behavior change in PR description and commit message, otherwise it's hard to track when this changes in the future.

Reply:
Got it, thanks.